cmake/LYPython.cmake

```diff
@@ -71,13 +82,43 @@ function(ly_setup_python_venv)
     endif()
 
     if (CREATE_NEW_VENV)
+        # Run the install venv command, but skip the pip setup because we may need to manually create 
+        # a link to the shared library within the created virtual environment before proceeding with
+        # the pip install command.
         message(STATUS "Creating Python venv at ${PYTHON_VENV_PATH}")
-        execute_process(COMMAND "${LY_3RDPARTY_PATH}/packages/${LY_PYTHON_PACKAGE_NAME}/${LY_PYTHON_EXECUTABLE}" -m venv "${PYTHON_VENV_PATH}"
-                        WORKING_DIRECTORY "${LY_3RDPARTY_PATH}/packages/${LY_PYTHON_PACKAGE_NAME}"
+        execute_process(COMMAND "${LY_3RDPARTY_PATH}/packages/${LY_PYTHON_PACKAGE_NAME}/${LY_PYTHON_BIN_PATH}/${LY_PYTHON_EXECUTABLE}" -m venv "${PYTHON_VENV_PATH}" --without-pip --clear
+                        WORKING_DIRECTORY "${LY_3RDPARTY_PATH}/packages/${LY_PYTHON_PACKAGE_NAME}/${LY_PYTHON_BIN_PATH}"
+                        COMMAND_ECHO STDOUT
                         RESULT_VARIABLE command_result)
+
         if (NOT ${command_result} EQUAL 0)
             message(FATAL_ERROR "Error creating a venv")
         endif()
+
+        if (LY_LINK_TO_SHARED_LIBRARY)
+            # Create a symlink to the package's python shared library inside the virtual environments
+            # own lib sub folder to match the original package structure
+            execute_process(COMMAND ln -s "${LY_3RDPARTY_PATH}/packages/${LY_PYTHON_PACKAGE_NAME}/${LY_PYTHON_LIB_PATH}/libpython3.10.so.1.0" libpython3.10.so.1.0
+                            WORKING_DIRECTORY "${PYTHON_VENV_PATH}/${LY_PYTHON_VENV_LIB_PATH}"
+                            COMMAND_ECHO STDOUT
+                            RESULT_VARIABLE command_result)
+            if (NOT ${command_result} EQUAL 0)
+                message(WARNING "Unable to createa venv shared library link.")
+            endif()
+        endif()
```

Comment:
FYI: For portability this logic might want to use the CMake `create_symlink` [command](https://cmake.org/cmake/help/latest/manual/cmake.1.html#cmdoption-cmake-E-arg-create_symlink).